### DISCUSSION TITLE
how to use canvas.getContext('2d') draw background

### DISCUSSION THREAD

**Discussion by kwdiwt** [fabricjs/fabric.js#8033] [Q&A] [answered]
I want to paint the background freely
example：
```
const t = canvas.getContext('2d')
t.restore
t.rotate
 .....
```

**Comment by ShaMan123** [comment, marked as answer]
You can:
1. Set an object as the `backgroundImage` (it can be any object)
2. hook to `before:render` 
3. override `_renderBackground`

 https://github.com/fabricjs/fabric.js/blob/8705f610ca19003ee0554574fccc6e90188f5491/src/static_canvas.class.js#L759
